src/ducks/transactions/TransactionCategoryEditor.jsx

```diff
@@ -9,32 +9,54 @@ import CategoryChoice from 'ducks/categories/CategoryChoice'
 /**
  * Edits a transaction's category through CategoryChoice
  */
-const TransactionCategoryEditor = props => {
+const TransactionCategoryEditor = ({
+  transactions,
+  beforeUpdate,
+  afterUpdate,
+  afterUpdates,
+  onCancel
+}) => {
   const client = useClient()
-  const { transaction, beforeUpdate, afterUpdate, onCancel } = props
+  const makeCategoryId = useMemo(() => getCategoryId(transactions[0]), [
+    transactions
+  ])
 
-  const handleSelect = async category => {
-    if (beforeUpdate) {
-      await beforeUpdate()
-    }
+  const handleUpdate = async (transaction, category) => {
     const newTransaction = await updateTransactionCategory(
       client,
       transaction,
       category
     )
+
     if (afterUpdate) {
       await afterUpdate(newTransaction)
     }
   }
 
+  const handleSelect = async category => {
+    if (beforeUpdate) {
+      await beforeUpdate()
+    }
+
+    const promises = transactions.map(transaction =>
+      handleUpdate(transaction, category)
+    )
+
+    await Promise.all(promises)
+
+    if (afterUpdates) {
+      afterUpdates()
+    }
+  }
```

Comment:
handlers could be `useCallback`ed here.

Reply:
j'en ai profité pour mémoiser le composant également